ompi/communicator/comm.c

```diff
@@ -1331,6 +1331,18 @@ int ompi_comm_create_from_group (ompi_group_t *group, const char *tag, opal_info
 
     newcomp->instance = group->grp_instance;
 
+    /*
+     * setup predefined keyvals - see MPI Standard for predefined keyvals cached on 
+     * communicators created via MPI_Comm_from_group or MPI_Intercomm_create_from_groups
+     */
+    ompi_attr_hash_init(&newcomp->c_keyhash);
+    ompi_attr_create_predefined_keyvals();
```

Comment:
Should this be invoked from MPI_Session_init?  It doesn't really matter much, but that would lead to it being invoked potentially fewer times over the course of an application (compared to being called each time a communicator is constructed from a group).